docs/advanced-topics/negative-mining.md

```diff
@@ -0,0 +1,91 @@
+(negative-mining)=
+# {octicon}`telescope` Negative Mining
+
+Negative Mining is an advanced technique, which optimizes the way data is sampled from you training dataset.
+Usually, it aims at making the metric learning tasks for the model harder during the training. 
+In this way it can lead to better fine-tuning results.
+
+## Context: Deep Metric Learning
+
+To understand negative mining, we first take a look at how the training data 
+construction is done for metric learning tasks.
+
+Metric Learning algorithms attempt to teach neural network models to tell
+which objects are semantically/visually similar and which ones are not.
+
+For uni-modal fine-tuning tasks such as text-to-text, image-to-image, or mesh-to-mesh,
+Finetuner constructs training data in the following way:
+
+![batch-sample](../imgs/batch-sampling.png)
+
+Assume we have a list of Documents belonging to four classes: `1`, `2`, `3`, and `4`,
+Finetuner will evenly sample *X* items per class to make a batch *B*.
+
+In the next step, arranges the items.
+Many loss functions used by Finetuner are calculated based on the embeddings of Triples. 
+For those, Finetuner creates all possible Triplets *(anchor, pos, neg)* from this batch.
+These Triplets become the training data.
```

Comment:
For each Triplet, the first is the __anchor__, the second is an embedding that ought to be closer to the embedding of the anchor, and the second is one that should be further from the anchor.